packages/table/src/Cell/HeaderCell/HeaderCell.tsx

```diff
@@ -33,16 +27,14 @@ const HeaderCell = <T extends LGRowData>({
   className,
   cellIndex,
   header,
+  align,
   ...rest
 }: PropsWithChildren<HeaderCellProps<T>>) => {
-  const { table } = useTableContext();
-
-  const isFirstCell = cellIndex === 0;
-  const isSelectable = !!table && !!table.hasSelectableRows;
+  const { isSelectable } = useTableContext();
 
   let columnName, sortState, onSortIconClick;
 
-  if (header && header.column.getCanSort()) {
+  if (header && header?.column.columnDef.enableSorting) {
     columnName = header.column.columnDef.header as string;
     const headerSortDirection = header.column.getIsSorted().toString();
     sortState = HeaderSortState[headerSortDirection];
```

Comment:
I'd create a utility for this 
```suggestion
  const {columnName, sortState, onSortIconClick} = getHeaderCellState(header)
```